### DISCUSSION TITLE
Linking essential libraries from AppImage.

### DISCUSSION THREAD

**Discussion by GamePlayer-8** [AppImage/AppImageKit#1261] [Q&A] [answered]
Heyo,

I have a problem with linking libraries to the application, which is using `musl` libc.

Even after including variables like `LD_LIBRARY_PATH` in `AppRun` it's still doesn't work. However running extracted AppImage in chroot works.

What variables should I additionally include in `AppRun` in order to make the AppImage work correctly?

Here's how my `AppRun` looks like:
```shell
#!/bin/sh
TICLOCK_RUNPATH="$(dirname "$(readlink -f "${0}")")"
TICLOCK_EXEC="${TICLOCK_RUNPATH}"/usr/bin/ticlock
LD_LIBRARY_PATH="${TICLOCK_RUNPATH}"/lib/:"${TICLOCK_RUNPATH}"/lib64/
LIBRARY_PATH="${TICLOCK_RUNPATH}"/lib/:"${TICLOCK_RUNPATH}"/lib64/
PATH=$PATH:"${TICLOCK_RUNPATH}"/bin/:"${TICLOCK_RUNPATH}"/sbin/:"${TICLOCK_RUNPATH}"/usr/bin/
exec "${TICLOCK_EXEC}" $@
```

The error what I'm getting is `No such file or directory` when running through AppImage.
I've added `alpine-base` package into AppImage.

**Comment by probonopd** [comment, marked as answer]
Instead of `exec "${TICLOCK_EXEC}" $@` you need to execute the musl libc ld loader, roughly along these lines:

```
${TICLOCK_RUNPATH}/lib/ld-musl-x86_64.so.1 "${TICLOCK_EXEC}" $@
```

I didn't test this but it should give you an idea.
